src/main/java/org/myrobotlab/service/InMoov2.java

```diff
@@ -596,82 +486,55 @@ public String get(String key) {
   }
 
   public InMoov2Arm getArm(String side) {
-    if ("left".equals(side)) {
-      return leftArm;
-    } else if ("right".equals(side)) {
-      return rightArm;
-    } else {
-      log.error("can not get arm {}", side);
-    }
-    return null;
+    return (InMoov2Arm) getPeer(side + "Arm");
   }
 
   public InMoov2Hand getHand(String side) {
-    if ("left".equals(side)) {
-      return leftHand;
-    } else if ("right".equals(side)) {
-      return rightHand;
-    } else {
-      log.error("can not get arm {}", side);
-    }
-    return null;
+    return (InMoov2Hand) getPeer(side + "Hand");
   }
 
   public InMoov2Head getHead() {
-    return head;
+    return (InMoov2Head) getPeer("head");
   }
 
   /**
    * finds most recent activity
    * 
    * @return the timestamp of the last activity time.
    */
-  public long getLastActivityTime() {
-
-    long lastActivityTime = 0;
-
-    if (leftHand != null) {
-      lastActivityTime = Math.max(lastActivityTime, leftHand.getLastActivityTime());
-    }
-
-    if (leftArm != null) {
-      lastActivityTime = Math.max(lastActivityTime, leftArm.getLastActivityTime());
-    }
-
-    if (rightHand != null) {
-      lastActivityTime = Math.max(lastActivityTime, rightHand.getLastActivityTime());
-    }
+  public Long getLastActivityTime() {
+    try {
 
-    if (rightArm != null) {
-      lastActivityTime = Math.max(lastActivityTime, rightArm.getLastActivityTime());
-    }
+      Long lastActivityTime = 0L;
 
-    if (head != null) {
-      lastActivityTime = Math.max(lastActivityTime, head.getLastActivityTime());
-    }
+      Long head = (Long) sendToPeerBlocking("head", "getScript", getName());
```

Comment:
what is the reason for sending a blocking call for "getScript" here ?  This seems odd/wrong .. the point of this method is to see the last time there was activity.

Reply:
copy paste error - good catch thanks ..

Reply:
updated to getLastActivityTime